### PR TITLE
REF-3 Home Screen Redesign

### PR DESCRIPTION
**Pull request by gmbugua**
Home Screen Updated UI

**Navigation Un-Linked**

Screen Shot 2020-06-23 at 9 24 37 PM

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ import SettingsEmailScreen from './src/components/auth/SettingsEmailScreen'
 import SettingsPasswordScreen from './src/components/auth/SettingsPasswordScreen'
 
 // MainStack
-import HomeScreen from './src/components/main/HomeScreen'
+import HomeScreen from './src/components/main/HomeScreen_alt'
```
Comment (wang1kevin1) [inline App.js:29]: If you're already replacing it, rename your file to HomeScreen.js and the old one to HomeScreen_Dep.js

File: src/components/main/HomeScreen_alt.js [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+
+    console.log(charsIata);
+    console.log(charsIcao);
+
+    let numsIata = spaceBuffer.slice(2);
+    let numsIcao = spaceBuffer.slice(3);
+
+    console.log(numsIata);
+    console.log(numsIcao);
+
```
Comment (jmarecaux) [inline src/components/main/HomeScreen_alt.js:97]: no error is ever thrown, we are taken to checkout with flight no matter what here

Comment (wang1kevin1) [reply src/components/main/HomeScreen_alt.js:97]: errors are thrown in iataCall and icaoCall

File: src/components/main/HomeScreen_alt.js [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+                
+                <View style={styles.appIntro}>
+
+                  <View style={styles.topText}>
+                    <Text style={styles.largeWhiteText}>A cada $6 uma</Text>
+                    <Text style={styles.largeWhiteText}>árvore é plantada.</Text>
+                  </View>
+
+                  <Text style={styles.mediumWhiteText}>
+                      {`Faça sua doação e ajude a recuperar a Mata Atlântica da Fazenda do Bulcão.`}
```
[the remark this answers is not in the document]
Comment (gmbugua) [reply src/components/main/HomeScreen_alt.js]: Multiline text => {``}

[the remark this answers is not in the document]
Comment (wang1kevin1) [reply src/components/main/HomeScreen_alt.js]: It should wrap itself without needing that. Note AboutScreen.js

[the remark this answers is not in the document]
Comment (gmbugua) [reply src/components/main/HomeScreen_alt.js]: White space is recognized when multiline syntax is not used.

[the remark this answers is not in the document]
Comment (gmbugua) [reply src/components/main/HomeScreen_alt.js]: Done! I changed it a while back and did not notice

File: src/components/main/HomeScreen_alt.js [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+                  </Text>
+
+                  <View style={styles.bottomText}> 
+                    <Text style={styles.smallWhiteText}>Insira o número de vôo para iniciar ou</Text>
+                    <Text style={styles.linkWhiteText}>doe sem número de vôo</Text>
```
[the remark this answers is not in the document]
Comment (gmbugua) [reply src/components/main/HomeScreen_alt.js]: I used '>>' text for now, however I think that is the double chevron icon. Which is not available using IONICONS

File: src/components/main/HomeScreen_alt.js [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+          this.setState({ error: true })
+          flightSearch.current.shake();
+          flightSearch.current.clear();
+
+        }
+      }).catch((error) => {
+        console.error(error);
+      });
+  }
+
```
[the remark this answers is not in the document]
Comment (gmbugua) [reply src/components/main/HomeScreen_alt.js]: Added, works, but there are some little problems like the starting position of elements and the ending. Also out of frame elements. However, I think that this was the intention. A scrolling animation
IMG_39A2ADCA6AA2-1


 of sorts.

[the remark this answers is not in the document]
Comment (gmbugua) [reply src/components/main/HomeScreen_alt.js]: I'm gonna push the current version and work on it more tomorrow.

File: src/components/main/HomeScreen_alt.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (wang1kevin1) [reply src/components/main/HomeScreen_alt.js]: You can use any one of these: https://oblador.github.io/react-native-vector-icons/

Search for 'right' and you should find them. Remember to import the package, and you just do pretty much the same thing except you call the new package.

I.e. <Feather ....

File: src/components/main/HomeScreen.js [hunk 6]
```diff
@@ -1,355 +1,377 @@
...
+
+    // Opens up email to bug-report
+    handleBugReports = () => {};
+
+    // Checks if a user is logged in
+    async checkAuth() {
+        await Auth.currentAuthenticatedUser({ bypassCache: true })
+            .then(() => {
+                console.log("A user is logged in");
+                this.setState({ isAuthenticated: true });
```
Comment (wang1kevin1) [inline src/components/main/HomeScreen.js:57]: No longer checking for auth on homescreen.

File: src/components/main/HomeScreen_alt.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (gmbugua) [reply src/components/main/HomeScreen_alt.js]: Also, is this a link? If so, where does it go?

File: src/components/main/HomeScreen_alt.js (hunk 2 above)
Comment (gmbugua) [reply src/components/main/HomeScreen_alt.js]: can you elaborate on this?

File: src/components/main/HomeScreen.js [hunk 7]
```diff
@@ -1,355 +1,325 @@
...
+                    source={this.background}
+                    style={styles.imageBackground}
+                >
+                    <KeyboardAvoidingView
+                        behavior={Platform.OS == "ios" ? "padding" : "height"}
+                        style={styles.container}
+                        enabled="false"
+                    >
+                        <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
+                            <View>
```
Comment (wang1kevin1) [inline src/components/main/HomeScreen.js:139]: TouchableWithoutFeedback should be outside of ImageBackground

File: src/components/main/MenuBar.js [hunk 8]
```diff
@@ -1,222 +1,224 @@
-import React from 'react'
+import React from "react";
```
Comment (wang1kevin1) [inline src/components/main/MenuBar.js:1]: what was changed here?

File: src/components/main/HomeScreen_alt.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (wang1kevin1) [reply src/components/main/HomeScreen_alt.js]: this goes to this.props.navigation.navigate("CheckoutWithoutFlight")

[the remark this answers is not in the document]
Comment (gmbugua) [reply src/components/main/HomeScreen_alt.js]: Nice, this is now resolved

[the remark this answers is not in the document]
Comment (gmbugua) [reply src/components/main/HomeScreen_alt.js]: I got the double chevron from importing MaterialCommunityIcons

File: src/components/main/MenuBar.js (hunk 8 above)
Comment (gmbugua) [reply src/components/main/MenuBar.js:1]: nothin just the qoutes

**Pull request closed without merging**